diff --git a/node_modules/@ai-sdk/google/dist/index.js b/node_modules/@ai-sdk/google/dist/index.js
index b708b4d..6b94db7 100644
--- a/node_modules/@ai-sdk/google/dist/index.js
+++ b/node_modules/@ai-sdk/google/dist/index.js
@@ -858,7 +858,7 @@ function prepareTools({
     "gemini-flash-lite-latest",
     "gemini-pro-latest"
   ].some((id) => id === modelId);
-  const isGemini2orNewer = modelId.includes("gemini-2") || modelId.includes("gemini-3") || modelId.includes("nano-banana") || isLatest;
+  const isGemini2orNewer = modelId.includes('gemma-4') || modelId.includes("gemini-2") || modelId.includes("gemini-3") || modelId.includes("nano-banana") || isLatest;
   const isGemini3orNewer = modelId.includes("gemini-3");
   const supportsFileSearch = modelId.includes("gemini-2.5") || modelId.includes("gemini-3");
   if (tools == null) {
diff --git a/node_modules/@ai-sdk/google/dist/index.mjs b/node_modules/@ai-sdk/google/dist/index.mjs
index 028f2de..c9fe739 100644
--- a/node_modules/@ai-sdk/google/dist/index.mjs
+++ b/node_modules/@ai-sdk/google/dist/index.mjs
@@ -866,7 +866,7 @@ function prepareTools({
     "gemini-flash-lite-latest",
     "gemini-pro-latest"
   ].some((id) => id === modelId);
-  const isGemini2orNewer = modelId.includes("gemini-2") || modelId.includes("gemini-3") || modelId.includes("nano-banana") || isLatest;
+  const isGemini2orNewer = modelId.includes('gemma-4') || modelId.includes("gemini-2") || modelId.includes("gemini-3") || modelId.includes("nano-banana") || isLatest;
   const isGemini3orNewer = modelId.includes("gemini-3");
   const supportsFileSearch = modelId.includes("gemini-2.5") || modelId.includes("gemini-3");
   if (tools == null) {
diff --git a/node_modules/@ai-sdk/google/dist/internal/index.js b/node_modules/@ai-sdk/google/dist/internal/index.js
index bfc039b..47313ca 100644
--- a/node_modules/@ai-sdk/google/dist/internal/index.js
+++ b/node_modules/@ai-sdk/google/dist/internal/index.js
@@ -666,7 +666,7 @@ function prepareTools({
     "gemini-flash-lite-latest",
     "gemini-pro-latest"
   ].some((id) => id === modelId);
-  const isGemini2orNewer = modelId.includes("gemini-2") || modelId.includes("gemini-3") || modelId.includes("nano-banana") || isLatest;
+  const isGemini2orNewer = modelId.includes('gemma-4') || modelId.includes("gemini-2") || modelId.includes("gemini-3") || modelId.includes("nano-banana") || isLatest;
   const isGemini3orNewer = modelId.includes("gemini-3");
   const supportsFileSearch = modelId.includes("gemini-2.5") || modelId.includes("gemini-3");
   if (tools == null) {
diff --git a/node_modules/@ai-sdk/google/dist/internal/index.mjs b/node_modules/@ai-sdk/google/dist/internal/index.mjs
index ba71d04..46c9243 100644
--- a/node_modules/@ai-sdk/google/dist/internal/index.mjs
+++ b/node_modules/@ai-sdk/google/dist/internal/index.mjs
@@ -655,7 +655,7 @@ function prepareTools({
     "gemini-flash-lite-latest",
     "gemini-pro-latest"
   ].some((id) => id === modelId);
-  const isGemini2orNewer = modelId.includes("gemini-2") || modelId.includes("gemini-3") || modelId.includes("nano-banana") || isLatest;
+  const isGemini2orNewer = modelId.includes('gemma-4') || modelId.includes("gemini-2") || modelId.includes("gemini-3") || modelId.includes("nano-banana") || isLatest;
   const isGemini3orNewer = modelId.includes("gemini-3");
   const supportsFileSearch = modelId.includes("gemini-2.5") || modelId.includes("gemini-3");
   if (tools == null) {
